src/main/java/org/semux/core/Wallet.java

```diff
@@ -120,9 +144,19 @@ public boolean unlock(String password) {
                     newAliases = readAddressAliases(key, dec);
                     break;
                 case 3:
-                    byte[] salt = dec.readBytes();
+                    salt = dec.readBytes();
+                    key = BCrypt.generate(Bytes.of(password), salt, BCRYPT_COST);
+                    newAccounts = readAccounts(key, dec, true, version);
+                    newAliases = readAddressAliases(key, dec);
+                    break;
+                case 4:
+                    salt = dec.readBytes();
                     key = BCrypt.generate(Bytes.of(password), salt, BCRYPT_COST);
+                    hdSeed = dec.readBytes();
+                    numHdAccounts = dec.readInt();
```

Comment:
What if I have an old copy of wallet with two keys, but in the meanwhile I have created few more?
I guess, I will not see those accounts when old wallet file copy is loaded, but creating "new address" will uncover, one by one, all previously used ones?